packages/http-specs/package.json

```diff
@@ -0,0 +1,54 @@
+{
+  "name": "@typespec/http-specs",
+  "version": "0.37.2",
+  "description": "Spec scenarios and mock apis",
+  "main": "dist/index.js",
+  "type": "module",
+  "scripts": {
+    "watch": "tsc -p ./tsconfig.build.json --watch",
+    "build": "tsc -p ./tsconfig.build.json",
+    "clean": "rimraf dist/ temp/",
+    "validate-scenarios": "spec validate-scenarios ./http",
+    "generate-scenarios-summary": "spec generate-scenarios-summary ./http",
+    "validate-mock-apis": "spec validate-mock-apis ./http",
```

Comment:
Does this pass in this PR?

Reply:
@timotheeguerin  The validate-mock-api has 2 sections. The scenario implementation is reporting all scenarios are implemented correctly:

```
C:\Users\sarajama\Projects\typespec\packages\spec-core>pnpm .\cmd\cli.mjs validate-mock-apis ..\http-specs\specs
info: result length: 52
info: Found 52 full scenarios.
info: Found 52 scenarios.
info: ✓ Scenario routes has all implemented mock apis.
info: ✓ Scenario special-words has all implemented mock apis.
info: ✓ Scenario authentication/api-key has all implemented mock apis.
info: ✓ Scenario authentication/union has all implemented mock apis.
info: ✓ Scenario authentication/oauth2 has all implemented mock apis.
info: ✓ Scenario encode/bytes has all implemented mock apis.
info: ✓ Scenario encode/datetime has all implemented mock apis.
info: ✓ Scenario encode/duration has all implemented mock apis.
info: ✓ Scenario parameters/basic has all implemented mock apis.
info: ✓ Scenario encode/numeric has all implemented mock apis.
info: ✓ Scenario parameters/body-optionality has all implemented mock apis.
info: ✓ Scenario parameters/collection-format has all implemented mock apis.
info: ✓ Scenario parameters/spread has all implemented mock apis.
info: ✓ Scenario payload/media-type has all implemented mock apis.
info: ✓ Scenario payload/json-merge-patch has all implemented mock apis.
info: ✓ Scenario payload/multipart has all implemented mock apis.
info: ✓ Scenario payload/xml has all implemented mock apis.
info: ✓ Scenario special-headers/conditional-request has all implemented mock apis.
info: ✓ Scenario special-headers/repeatability has all implemented mock apis.
info: ✓ Scenario type/array has all implemented mock apis.
info: ✓ Scenario type/dictionary has all implemented mock apis.
info: ✓ Scenario type/scalar has all implemented mock apis.
info: ✓ Scenario type/union has all implemented mock apis.
info: ✓ Scenario versioning/added has all implemented mock apis.
info: ✓ Scenario versioning/madeOptional has all implemented mock apis.
info: ✓ Scenario versioning/removed has all implemented mock apis.
info: ✓ Scenario versioning/renamedFrom has all implemented mock apis.
info: ✓ Scenario versioning/returnTypeChangedFrom has all implemented mock apis.
info: ✓ Scenario versioning/typeChangedFrom has all implemented mock apis.
info: ✓ Scenario authentication/http/custom has all implemented mock apis.
info: ✓ Scenario server/endpoint/not-defined has all implemented mock apis.
info: ✓ Scenario serialization/encoded-name/json has all implemented mock apis.
info: ✓ Scenario server/path/multiple has all implemented mock apis.
info: ✓ Scenario server/path/single has all implemented mock apis.
info: ✓ Scenario server/versions/versioned has all implemented mock apis.
info: ✓ Scenario type/enum/extensible has all implemented mock apis.
info: ✓ Scenario server/versions/not-versioned has all implemented mock apis.
info: ✓ Scenario type/model/empty has all implemented mock apis.
info: ✓ Scenario type/enum/fixed has all implemented mock apis.
info: ✓ Scenario type/model/templated has all implemented mock apis.
info: ✓ Scenario type/model/usage has all implemented mock apis.
info: ✓ Scenario type/model/visibility has all implemented mock apis.
info: ✓ Scenario type/property/additional-properties has all implemented mock apis.
info: ✓ Scenario type/property/optionality has all implemented mock apis.
info: ✓ Scenario type/property/nullable has all implemented mock apis.
info: ✓ Scenario type/property/nullable has all implemented mock apis.
info: ✓ Scenario type/property/value-types has all implemented mock apis.
info: ✓ Scenario type/model/inheritance/not-discriminated has all implemented mock apis.
info: ✓ Scenario type/model/inheritance/enum-discriminator has all implemented mock apis.
info: ✓ Scenario type/model/inheritance/nested-discriminator has all implemented mock apis.
info: ✓ Scenario type/model/inheritance/recursive has all implemented mock apis.
info: ✓ Scenario type/model/inheritance/single-discriminator has all implemented mock apis. 
```

But, the program diagnostics is reporting errors as:

```
C:\Users\sarajama\Projects\typespec\packages\spec-core>pnpm .\cmd\cli.mjs validate-mock-apis ..\http-specs\specs
info: result length: 52
info: Found 52 full scenarios.
info: Found 52 scenarios.
error: ✘ Scenario routes is invalid.
error: ✘ Scenario special-words is invalid.
error: ✘ Scenario authentication/api-key is invalid.
error: ✘ Scenario authentication/oauth2 is invalid.
error: ✘ Scenario authentication/union is invalid.
error: ✘ Scenario encode/bytes is invalid.
error: ✘ Scenario encode/datetime is invalid.
error: ✘ Scenario encode/duration is invalid.
error: ✘ Scenario parameters/basic is invalid.
error: ✘ Scenario encode/numeric is invalid.
error: ✘ Scenario parameters/body-optionality is invalid.
error: ✘ Scenario parameters/collection-format is invalid.
error: ✘ Scenario parameters/spread is invalid.
error: ✘ Scenario payload/json-merge-patch is invalid.
error: ✘ Scenario payload/media-type is invalid.
error: ✘ Scenario payload/xml is invalid.
error: ✘ Scenario payload/multipart is invalid.
error: ✘ Scenario special-headers/conditional-request is invalid.
error: ✘ Scenario special-headers/repeatability is invalid.
error: ✘ Scenario type/array is invalid.
error: ✘ Scenario type/dictionary is invalid.
error: ✘ Scenario type/scalar is invalid.
error: ✘ Scenario type/union is invalid.
error: ✘ Scenario versioning/added is invalid.
error: ✘ Scenario versioning/madeOptional is invalid.
error: ✘ Scenario versioning/removed is invalid.
error: ✘ Scenario versioning/renamedFrom is invalid.
error: ✘ Scenario versioning/returnTypeChangedFrom is invalid.
error: ✘ Scenario versioning/typeChangedFrom is invalid.
error: ✘ Scenario authentication/http/custom is invalid.
error: ✘ Scenario serialization/encoded-name/json is invalid.
error: ✘ Scenario server/endpoint/not-defined is invalid.
error: ✘ Scenario server/path/multiple is invalid.
error: ✘ Scenario server/path/single is invalid.
error: ✘ Scenario server/versions/not-versioned is invalid.
error: ✘ Scenario server/versions/versioned is invalid.
error: ✘ Scenario type/enum/extensible is invalid.
error: ✘ Scenario type/model/empty is invalid.
error: ✘ Scenario type/enum/fixed is invalid.
error: ✘ Scenario type/model/templated is invalid.
error: ✘ Scenario type/model/usage is invalid.
error: ✘ Scenario type/model/visibility is invalid.
error: ✘ Scenario type/property/additional-properties is invalid.
error: ✘ Scenario type/property/nullable is invalid.
error: ✘ Scenario type/property/nullable is invalid.
error: ✘ Scenario type/property/optionality is invalid.
error: ✘ Scenario type/property/value-types is invalid.
error: ✘ Scenario type/model/inheritance/enum-discriminator is invalid.
error: ✘ Scenario type/model/inheritance/nested-discriminator is invalid.
error: ✘ Scenario type/model/inheritance/not-discriminated is invalid.
error: ✘ Scenario type/model/inheritance/recursive is invalid.
error: ✘ Scenario type/model/inheritance/single-discriminator is invalid.
```

Reply:
yeah so that probably mean that those scenarios don't have the correct mapping tor the mock api. Make sure the name match

Reply:
I have matched all the scenarios.  The naming is correct for all the scenarios

Reply:
@timotheeguerin I have double checked the scenario names and also cross checked them one by one with cadl-ranch-specs repository also. The naming is correct. Also, as you see in the above output, the scenario count is correct. So, it is not the missing scenarios. But, the system is not able  to identify `@scenario`, `@scenarioService`, etc. In VS code, the editor is able to identify them. But, when I run the tool, it is not able to identify.